controllers/glanceapi_controller.go

```diff
@@ -120,6 +120,27 @@ func (r *GlanceAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, nil
 	}
 
+	// Get Glance Top level CR
+	parentGlance := glance.GetOwningGlanceName(instance)
+	r.Log.Info(fmt.Sprintf("Glance Top Level CR found %s", parentGlance))
+
+	glance := &glancev1.GlanceList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(req.Namespace),
+	}
+	if err := r.Client.List(context.Background(), glance, listOpts...); err != nil {
+		r.Log.Error(err, "Unable to retrieve Glance CR %v")
+		return ctrl.Result{}, nil
+	}
+	for _, cr := range glance.Items {
+		// Filter the Glance CR by kind and name
+		if cr.Kind == "Glance" && cr.Name == parentGlance {
+			r.Log.Info(fmt.Sprintf("Glance Top Level CR found %s", cr.Name))
+			r.Log.Info(fmt.Sprintf("ExtraVolumes Defined in the Top Level CR %v", cr.Spec.ExtraMounts))
+			instance.ExtraMounts = cr.Spec.ExtraMounts
```

Comment:
@abays so basically here I'm doing something similar to what `Cinder` does: I watch/get the top-level (parent) CR and I fetch the `ExtraMounts` (if not defined, `instance.ExtraMounts` is not assigned).

Reply:
I'm curious as to why we would get a list here as opposed to just getting the `Glance` CR directly, since we have its name and namespace.  i.e. why not do something like this [1] instead (old example from OSPdO but demonstrates the approach nonetheless)? 

[1] https://github.com/openstack-k8s-operators/osp-director-operator/blob/3d651137b62cfe7ff937c8ed5d04638f3194282a/api/v1beta1/common_openstacknet.go#L213-L227

Reply:
I was thinking more at this use case, you can technically have more than one `Glance` CR deployed, but the relationship with the sub-resources is established in the controller, so I don't have to list all the `Glance` related `CRs` and watch for the right one to fetch `data`: I basically created that part starting from a wrong assumption.
Thanks actually for pointing me to [1], it's helpful and I'm pushing the same approach described there.

[1] https://github.com/openstack-k8s-operators/osp-director-operator/blob/3d651137b62cfe7ff937c8ed5d04638f3194282a/api/v1beta1/common_openstacknet.go#L213-L227